distributed/core.py

```diff
@@ -98,6 +98,11 @@ def _raise(*args, **kwargs):
 LOG_PDB = dask.config.get("distributed.admin.pdb-on-err")
 
 
+def _expects_comm(func):
+    sig = inspect.signature(func)
+    return "comm" in sig.parameters or "stream" in sig.parameters
```

Comment:
Where is `stream` used? Is it for potential third party scheduler extensions? If so, I think there should be a deprecation cycle.

I could not find any internal use of stream, besides this one which you forgot to remove:
https://github.com/fjetter/distributed/blob/a16c7e713a787b52573c31664db98d2fb275dab7/distributed/publish.py#L50-L52

Reply:
How about we raise a deprecation warning here if we hit stream? I noticed there is another place and didn't clean it up since I wasn't sure about the actual usage.

Note: The change I am proposing here is technically already a breaking change to user plugins. If they are not using any of the two names, their plugin might break. 

I'm wondering if this warrants a deprecation cycle as well. 

Thoughts?

Reply:
I think the deprecation system you put in place is already plenty